### DISCUSSION TITLE
Improve documentation of `chroot`

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [dompdf/dompdf#2346] [General]
Currently, there are three locations where `chroot` is explained:
* https://github.com/dompdf/dompdf/wiki/Usage#security-restrictions-for-local-files
* https://github.com/dompdf/dompdf/wiki/Usage#options
* https://github.com/dompdf/dompdf/blob/v1.0.1/src/Options.php#L46..L63

These should be merged to one ;-)

Besides, all dompdf code samples should include a line like this to make people aware of the importance of that setting:
```php
$dompdf->options->setChroot(__DIR__);
```

The fact that I did not want to include such a line on every dompdf call in my code, made me start using a Symfony bundle, and I created a composer recipe for it which sets `chroot` to a sensible default: https://github.com/symfony/recipes-contrib/pull/1075 Since probably most dompdf users are using it inside some framework, you should think about creating a list of all available framework integrations in `REAMDE.md`.

**Comment by bsweeney** [comment]
The code comment is to provide method documentation for IDEs and applications like phpdoc, so that should be treated separately from the wiki. I think it would be useful to have the wiki content auto-generate. I'm working on auto generating API documentation from the code documentation, so that should help.

And though we want to avoid duplication of content, I do see some value in calling out some information that may be of particular interest to users. So while the options definitions should be auto-generated I do believe additional explanation related to resources references is appropriate.

I also agree that having to set all values on each call can be ... onerous. It's why I generally extract Dompdf calls to a helper method in my own usage. I think adding framework-related documentation with references to relevant project would be useful to have in the wiki and encourage anyone to do so (otherwise what's the point of a wiki). I'm not involved in any such integrations so I don't have a lot of information to share.
